modules/Header/ui/CategoriesNav/CategoriesNavDesktop.tsx

```diff
@@ -127,24 +140,34 @@ export function CategoriesNavDesktop({
                                 <div className={styles.backdrop} />
                             </div>
                         </Transition>
-                        {open && <BlockPageScroll />}
+                        {open && <BlockPageScroll currentWidth={currentWidth} />}
                     </>
                 )}
             </Popover>
         </li>
     );
 }
 
-function BlockPageScroll() {
+function BlockPageScroll({ currentWidth }: BlockPageScroll.Props) {
     useEffect(() => {
         document.body.classList.add(styles.preventScroll);
-
-        return () => document.body.classList.remove(styles.preventScroll);
-    }, []);
+        const scrollBarWidth = document.body.offsetWidth - currentWidth;
+        document.body.style.marginRight = `${scrollBarWidth}px`;
+        return () => {
+            document.body.classList.remove(styles.preventScroll);
+            document.body.style.marginRight = `0`;
+        };
+    });
 
     return null;
 }
 
+export namespace BlockPageScroll {
+    export interface Props {
+        currentWidth: number;
+    }
+}
+
```

Comment:
This can be removed after applying my suggestion on moving the effect & state to `BlockPageScroll`.